### PR TITLE
0.2.0 Ruler functionality

### PR DESCRIPTION
**Pull request by valerii-cognite**
This is a child PR of https://github.com/cognitedata/griff-react/pull/47

### CODE REVIEW

File: src/components/InteractionLayer/index.js [hunk 1]
```diff
@@ -106,50 +104,17 @@ class InteractionLayer extends React.Component {
...
-            id: s.id,
-            timestamp: ts,
-            value,
-            x: xScale(ts),
-            y: yScale(value),
-          });
-        } else {
-          output.points.push({ id: s.id });
-        }
+    if (onMouseMove || ruler) {
```
Comment (evancharlton) [inline src/components/InteractionLayer/index.js]: This fixes #24 -- don't forget to put that in your commit!

File: src/components/Ruler/index.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    }
+    pointsObject[pointsSorted[i].name] = prevPoint;
+  }
+
+  return pointsObject;
+};
+
+const Ruler = ({ ruler, points, height, width }) => {
+  const xTooltipMargin = 5;
+  const xTooltipPosition = height - labelHeight - xTooltipMargin;
```
Comment (evancharlton) [inline src/components/Ruler/index.js]: Now it's called `xTooltipPosition`? I'm even more confused.

What does this value actually represent? As far as I can tell, it's just a point 29px (`labelHeight + xTooltipMargin`) off the bottom (`height - 29`) of the chart, but I don't know why, or what it's used for?

**Comment by evancharlton** [comment]
I get this in the storybook: 
```
warning.js:33 Warning: Failed prop type: The prop `ruler` is marked as required in `InteractionLayer`, but its value is `undefined`.
```

File: src/components/InteractionLayer/index.js (hunk 1 above)
Comment (valerii-cognite) [reply src/components/InteractionLayer/index.js]: Good catch

File: src/components/Ruler/index.js (hunk 2 above)
Comment (valerii-cognite) [reply src/components/Ruler/index.js]: ok, just added a comment

**Comment by valerii-cognite** [comment]
Missed proper propTypes, fixed

File: src/components/RulerTooltip/index.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+};
+
+class RulerTooltip extends Component {
+  state = {
+    textWidth: 0,
+  };
+
+  componentDidMount() {
+    const textBBox = this.textRef.getBBox();
+    // eslint-disable-next-line react/no-did-mount-set-state
```
Comment (evancharlton) [inline src/components/RulerTooltip/index.js]: Why are we able to ignore this?

Comment (martincognite) [reply src/components/RulerTooltip/index.js]: Is this a `setState` in did/will mount discussion?

Comment (valerii-cognite) [reply src/components/RulerTooltip/index.js]: It seems the second render is not an issue [anymore](https://github.com/airbnb/javascript/issues/684#issuecomment-381674576), so airbnb want to turn this error off. The other problem we have I didn't think about before is that we can't really know in advance the text width has been already set in componentDidMount hook, but as far as it is synchronous operation we can run setState right after all the sync operations finish by wrapping it in an immediate setTimeout, i.e.
```
setTimeout(() =>
  this.setState({
    textWidth: textBBox.width,
  })
);
```

**Pull request merged**